errors.go

```diff
@@ -335,3 +335,29 @@ func Find(origErr error, test func(error) bool) error {
 	})
 	return foundErr
 }
+
+// Chain ties together two errors, giving them a Causer interface and a concatenated Error message.
+type chain struct {
+	currentErr error
+	prevErr    error
+}
+
+func (err *chain) Error() string {
+	return err.currentErr.Error() + "\n" + err.prevErr.Error()
+}
+
+func (err *chain) Cause() error {
+	return err.prevErr
```

Comment:
should we recursively find the root cause?

Reply:
No: this is the `causer` interface that goes just one level deeper.